pkg/sentry/fsimpl/devpts/line_discipline.go

```diff
@@ -212,9 +212,14 @@ func (l *lineDiscipline) inputQueueRead(ctx context.Context, dst usermem.IOSeque
 			l.replicaWaiter.Notify(waiter.ReadableEvents)
 		}
 		return n, nil
-	} else if notifyEcho {
+	}
+	if notifyEcho {
 		l.masterWaiter.Notify(waiter.ReadableEvents)
 	}
+	if !pushed && l.termios.LEnabled(linux.ICANON) {
```

Comment:
We have to hold `l.termiosMu` to access `l.termios`.

I would set `isCanon := t.ermios.LEnabled(linux.ICANON)`  while the lock is held on line 200, and then reference `isCanon` in this conditional.